ansibullbot/triagers/plugins/docs_info.py

```diff
@@ -0,0 +1,306 @@
+                    )
+
+            for child_node in ast.iter_child_nodes(node):
+                if isinstance(child_node, (ast.FunctionDef, ast.AsyncFunctionDef)):
+                    func_map = ParsedFunc(
+                        name=child_node.name,
+                        line_start=child_node.lineno,
+                        line_end=child_node.end_lineno,
+                    )
+                    func_doc = ast.get_docstring(child_node)
+                    if func_doc is not None:
+                        func_map.doc_string = func_doc
+                        ds_span = re.search(re.escape(func_doc), content)
+                        if ds_span is not None:
+                            func_map.ds_line_start = len(
+                                content[:ds_span.start()].splitlines()
+                            )
+                            func_map.ds_line_end = (
+                                func_map.ds_line_start + len(func_doc.splitlines())
+                        )
+                    class_map.funcs.append(func_map)
+
+            mod_map.classes.append(class_map)
+
+    return mod_map
+
+def _check_py_changes(file_content, diff):
+    """ Check a python file's changes to see if they're only docstring
+        changes.
+    """
+
+    diff = _get_diff_info(diff)
+    source = None
+
+    if file_content is not None:
+        source = _get_ast_info(file_content)
+
+    if source is not None:
+        for line in diff:
+            # Check if change applies to module DOCUMENTATION (if exists)
+            if source.doc_string:
+                if source.ds_line_start <= line["lineno"] <= source.ds_line_end:
+                    continue
+
+            # Check if change applies to module EXAMPLES (if exists)
+            if source.example_string:
+                if source.ex_line_start <= line["lineno"] <= source.ex_line_end:
+                    continue
+
+            # Find appropriate class if it exists
+            source_class = source.find_class(line["lineno"])
+            if source_class is not None:
+                if source_class.doc_string:
+                    if source_class.ds_line_start <= line["lineno"] <= source_class.ds_line_end:
+                        continue
+
+                source_func = source_class.find_function(line["lineno"])
+                if source_func is not None and source_func.doc_string:
+                    if source_func.ds_line_start <= line["lineno"] <= source_func.ds_line_end:
+                        continue
+
+            # If we made it this far, this change is outside docs/examples
+            return False
+
+        return True
+
+def _is_docs_only(changed_file):
+    """ Check if the changes made to ``changed_file`` affect only documentation. """
+    docs_only = False
+
+    if isinstance(changed_file, dict):
+        changed_file = CommitFile(changed_file)
+
+    if _is_docs_path(changed_file.filename):
+        docs_only = True
```

Comment:
I think if we just `return` here and do the same for the rest of the function we can get rid of all `else`s and reduce the complexity here.

Reply:
So we would end up with:
```python
def _is_docs_only(changed_file):
    """ Check if the changes made to ``changed_file`` affect only documentation. """
    if isinstance(changed_file, dict):
        changed_file = CommitFile(changed_file)

    if _is_docs_path(changed_file.filename):
        return True

    # Additions or deletions of complete files outside of
    # docsite/examples will likely be more than documentation
    # changes
    if changed_file.status != "modified":
        return False

    # Non-Python files that are outside of the docsite
    # folder will not be documentation related
    if not changed_file.filename.endswith(".py"):
        return False

    # If a python file, check if the changes are only to
    # docstrings
    if _check_py_changes(changed_file.file_content, changed_file.patch):
        return True

    return False
```

Reply:
Hrm, and we can go further and just do:
```python
def _is_docs_only(changed_file):
    """ Check if the changes made to ``changed_file`` affect only documentation. """
    if isinstance(changed_file, dict):
        changed_file = CommitFile(changed_file)

    return (
        _is_docs_path(changed_file.filename) or _check_py_changes(changed_file.file_content, changed_file.patch)
    )
```
?

Reply:
I went with the longer form, to preserve the intent comments.